### PR TITLE
Add Makefile targets for arm64

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Add `Makefile` targets for `arm64`.

**Why?**
To support build of cross-platform docker images.

**How did you test it?**
Run and build images locally.

**Potential risks**
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

**Review by wxing1292 — APPROVED** [review]

**Review by yiminc — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -480,6 +480,25 @@ docker-admin-tools:
...
+
+docker-server-x:
+	@printf $(COLOR) "Building cross-platform docker image temporalio/server:$(DOCKER_IMAGE_TAG)..."
+	docker buildx build . -t temporalio/server:$(DOCKER_IMAGE_TAG) --platform linux/amd64,linux/arm64 --output type=image --build-arg TARGET=server
+
+docker-auto-setup-x:
+	@printf $(COLOR) "Build cross-platform docker image temporalio/auto-setup:$(DOCKER_IMAGE_TAG)..."
+	docker buildx build . -t temporalio/auto-setup:$(DOCKER_IMAGE_TAG) --platform linux/amd64,linux/arm64 --output type=image --build-arg TARGET=auto-setup
+
+docker-tctl-x:
```
Comment (dnr) [inline Makefile:494]: Should we still include tctl targets? That's in another repo now, right? (which will also need changes to be built for arm64)

File: Makefile [hunk 2]
```diff
@@ -480,6 +480,25 @@ docker-admin-tools:
 	@printf $(COLOR) "Build docker image temporalio/admin-tools:$(DOCKER_IMAGE_TAG)..."
 	docker build . -t temporalio/admin-tools:$(DOCKER_IMAGE_TAG) --build-arg TARGET=admin-tools
 
+docker-buildx-container:
```
Comment (dnr) [inline Makefile:483]: What's the point of this? Do I need to do it before using the other targets? Maybe a comment would be nice...

File: docker/base-images/Makefile [hunk 3]
```diff
@@ -20,17 +20,17 @@ base-ci-builder:
 	docker push temporalio/base-ci-builder:$(DOCKER_IMAGE_TAG)
 
 ##### docker buildx #####
-buildx-docker-container:
+docker-buildx-container:
 	docker buildx create --driver docker-container --use
 
 base-admin-tools-x:
```
Comment (dnr) [inline docker/base-images/Makefile:26]: Why do these auto-push (`type=registry`) if the server ones don't? If you want to build locally, you can just build base images and then server images all locally, right? Pushing should be explicit, I think (maybe one make target to push everything?)

File: Makefile (hunk 1 above)
Comment (yiminc) [reply Makefile:494]: That one is not fully ready yet.

File: docker/base-images/Makefile (hunk 3 above)
Comment (alexshtin) [reply docker/base-images/Makefile:26]: Added variable.

File: Makefile (hunk 2 above)
Comment (alexshtin) [reply Makefile:483]: Yes, this target needs to be run once to create build container. I don't want to pollute `Makefile` with comments on how to use it. I would better add some readme for this later.

**Pull request merged**